File: backend/test_api.py
"""
Basic API tests to verify endpoints are working correctly
"""
import requests
import uuid

BASE_URL = "http://localhost:8000"

def test_health_endpoint():
    """Test the health check endpoint"""
    response = requests.get(f"{BASE_URL}/health")
    assert response.status_code == 200
    data = response.json()
    assert data["status"] == "healthy"
    print("✓ Health endpoint test passed")

def test_root_endpoint():
    """Test the root endpoint"""
    response = requests.get(f"{BASE_URL}/")
    assert response.status_code == 200
    data = response.json()
    assert "message" in data
    print("✓ Root endpoint test passed")

def test_api_structure():
    """Test that API endpoints exist (without authentication)"""
    # Test auth endpoints exist
    try:
        # This will fail due to missing auth, but should return 401/403 not 404
        response = requests.get(f"{BASE_URL}/api/auth/signup")
        # If we get here, endpoint exists (though it may not accept GET)
        print(f"✓ Auth endpoint structure exists (status: {response.status_code})")
    except:
        print("✓ Auth endpoints exist (would require proper method/params)")

if __name__ == "__main__":
    print("Running basic API tests...")
    try:
        test_health_endpoint()
        test_root_endpoint()
        test_api_structure()
        print("\nAll basic API tests completed!")
    except Exception as e:
        print(f"Test failed with error: {e}")

File: test_api.js
const axios = require('axios');

// Simple test to verify the API endpoints are accessible
async function testApiEndpoints() {
  try {
    // Test the root endpoint
    const rootResponse = await axios.get('http://localhost:8000/');
    console.log('Root endpoint status:', rootResponse.status);

    // Test the health endpoint
    const healthResponse = await axios.get('http://localhost:8000/health');
    console.log('Health endpoint status:', healthResponse.status);

    // Test the chat endpoint (should return 422 since we're not sending proper data)
    try {
      const chatResponse = await axios.post('http://localhost:8000/api/v1/chat/', {});
    } catch (error) {
      if (error.response && error.response.status === 422) {
        console.log('Chat endpoint accessible (expected 422 for missing data)');
      } else {
        console.error('Unexpected error calling chat endpoint:', error.message);
      }
    }

    console.log('API endpoints are accessible');
  } catch (error) {
    console.error('Error testing API endpoints:', error.message);
  }
}

testApiEndpoints();